test/factor/factorQuadratic.test.js

```diff
@@ -46,3 +46,64 @@ describe('factorQuadratic', function () {
   ];
   tests.forEach(t => testFactorQuadratic(t[0], t[1]));
 });
+
+function testFactorSumProductRuleSubsteps(exprString, outputList) {
+  const lastString = outputList[outputList.length - 1];
+  TestUtil.testSubsteps(factorQuadratic, exprString, outputList, lastString);
+}
+
+describe('factorSumProductRule', function() {
+  const tests = [
+    // sum product rule
+    ['x^2 + 3x + 2',
+      ['x^2 + x + 2x + 2',
+        '(x^2 + x) + (2x + 2)',
+        'x * (x + 1) + (2x + 2)',
+        'x * (x + 1) + 2 * (x + 1)',
+        '(x + 1) * (x + 2)']
+    ],
+    ['x^2 - 3x + 2',
+      ['x^2 - x - 2x + 2',
+        '(x^2 - x) + (-2x + 2)',
+        'x * (x - 1) + (-2x + 2)',
+        'x * (x - 1) - 2 * (x - 1)',
+        '(x - 1) * (x - 2)']
+    ],
+    ['x^2 + x - 2',
+      ['x^2 - x + 2x - 2',
+        '(x^2 - x) + (2x - 2)',
+        'x * (x - 1) + (2x - 2)',
+        'x * (x - 1) + 2 * (x - 1)',
+        '(x - 1) * (x + 2)']
+    ],
+    ['-x^2 - 3x - 2',
+      ['x^2 + x + 2x + 2', // per factorQuadratic, negative is added back in at the end
```

Comment:
Yes that makes much more sense. 😂  Looks I just had to negate each ```newNode``` along the way.